### PR TITLE
drivers: CAN: Add CAN shell

### PR DESCRIPTION
**Pull request by alexanderwachter**
This PR adds a CAN shell for easy testing and bus-sniffing.

### CODE REVIEW

File: drivers/can/can_shell.c [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misc/printk.h>
+#include <shell/shell.h>
+#include <drivers/can.h>
+#include <zephyr/types.h>
+#include <stdlib.h>
+
+struct zcan_work work;
```
Comment (jukkar) [inline drivers/can/can_shell.c]: static missing?

Comment (alexanderwachter) [reply drivers/can/can_shell.c]: Yes.  static is added now.

**Review by jukkar — APPROVED** [review]

**Review by tbursztyka — DISMISSED** [review]
drivers/can is the wrong location for a shell module. Find something else. maybe subsys/net/l2/canbus ?

**Comment by alexanderwachter** [comment]
@tbursztyka This shell module is only for the CAN driver and has nothing to do with networking.
Other examples for shell modules located in the driver directory: hwinfo, gpio, adc.

**Comment by tbursztyka** [comment]
@alexanderwachter ah ok, sounds weird. fine then

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Nice work, thanks. I think we should enable the CAN shell in `tests/boards/board_shell` as well.

**Comment by alexanderwachter** [comment]
@henrikbrixandersen sure. Didn't knew that we have tests for shell modules.

**Comment by jukkar** [comment]
As this CAN shell deals with the device names, it might be useful if we would activate also CONFIG_DEVICE_SHELL in CAN shell Kconfig as otherwise user might have some difficulty figuring out the CAN device name(s). No need to do it in this PR, this was just something I noticed.

**Comment by alexanderwachter** [comment]
@jukkar good catch! I'll add it in a second.

File: tests/boards/board_shell/prj.conf [hunk 2]
```diff
@@ -19,3 +19,5 @@ CONFIG_FLASH_MAP=y
 CONFIG_FLASH_MAP_SHELL=y
 CONFIG_ADC=y
 CONFIG_ADC_SHELL=y
+CONFIG_CAN=y
+CONFIG_CAN_SHELL=y
```
Comment (jukkar) [inline tests/boards/board_shell/prj.conf]: newline is missing

File: samples/drivers/CAN/prj.conf [hunk 3]
```diff
@@ -2,3 +2,7 @@ CONFIG_CAN=y
 CONFIG_CAN_INIT_PRIORITY=80
 CONFIG_CAN_1=y
 CONFIG_CAN_MAX_FILTER=5
+
+CONFIG_SHELL=y
+CONFIG_CAN_SHELL=y
+CONFIG_DEVICE_SHELL=y
```
Comment (jukkar) [inline samples/drivers/CAN/prj.conf]: newline is missing

File: drivers/can/Kconfig [hunk 4]
```diff
@@ -20,6 +20,12 @@ module = CAN
 module-str = CAN
 source "subsys/logging/Kconfig.template.log_config"
 
+config CAN_SHELL
+	bool "Enable CAN Shell"
+	depends on SHELL
+	help
```
Comment (jukkar) [inline drivers/can/Kconfig]: About the device shell, what about enabling it here unconditionally like this
`select DEVICE_SHELL`

Comment (alexanderwachter) [reply drivers/can/Kconfig]: Since the device-shell is not necessarily needed, I won't.

Comment (jukkar) [reply drivers/can/Kconfig]: Ok, that is fine, just pondering about it.

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**